src/main/java/com/example/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -4,8 +4,10 @@
 import com.example.onlinebookstore.repository.BookRepository;
 import jakarta.persistence.EntityManager;
 import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityNotFoundException;
```

Comment:
You have custom com.example.onlinebookstore.exception.EntityNotFoundException;
In other files you use this one. Maybe it is a part of some logic that I don't get

Reply:
Yeah, tnx for advice, its my mistake. Chose wrong library (same exception in jakarta package instead of custom).
Replace with Runtime atm.